File: model-service/src/model/model.controller.ts
import { Controller, Get, NotFoundException, Param } from '@nestjs/common';
import { ModelService } from './model.service';
import { Model } from './model';

@Controller('models')
export class ModelController {
  constructor(private modelService: ModelService) {}
  @Get('/:id')
  findModelById(@Param('id') id: string): Model {
    const result = this.modelService.findModelById(id);
    if (!result) {
      throw new NotFoundException(`model ${id} not found`);
    }
    return result;
  }
  @Get()
  findAllModels(): Model[] {
    return this.modelService.findAllModels();
  }
}

File: execution-service/model/model_controller.go
package model

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type ModelController struct {
	modelService ModelService
}

func NewModelController(modelService ModelService) ModelController {
	return ModelController{
		modelService: modelService,
	}
}

func (controller *ModelController) GetModel(c *gin.Context) {
	modelId := c.Param("modelId")
	model, err := controller.modelService.FindModelById(modelId)
	if err != nil {
		return
	}
	c.IndentedJSON(http.StatusOK, ToJson(model))
}

func (controller *ModelController) PutModel(c *gin.Context) {
	var modelJson ModelJson
	if err := c.BindJSON(&modelJson); err != nil {
		return
	}
	model, err := modelJson.ToModel()
	if err != nil {
		return
	}
	controller.modelService.SaveModel(model)
	c.Status(http.StatusOK)
}
